src/python/pants/engine/streaming_workunit_handler.py

```diff
@@ -102,23 +102,24 @@ def get_expanded_specs(self) -> ExpandedSpecs:
         """Return a dict containing the canonicalized addresses of the specs for this run, and what
         files they expand to."""
 
-        (unexpanded_addresses,) = self._scheduler.product_request(
-            Addresses, [Params(self._specs, self._options_bootstrapper)]
-        )
+        params = Params(self._specs, self._options_bootstrapper)
+        request = self._scheduler.execution_request([(Addresses, params), (Targets, params)])
+        unexpanded_addresses, expanded_targets = self._scheduler.execute(request)
 
-        expanded_targets = self._scheduler.product_request(
-            Targets, [Params(Addresses([addr])) for addr in unexpanded_addresses]
-        )
-        targets_dict: dict[str, list[TargetInfo]] = {}
-        for addr, targets in zip(unexpanded_addresses, expanded_targets):
-            targets_dict[addr.spec] = [
+        targets_dict: Dict[str, list[TargetInfo]] = {str(addr): [] for addr in unexpanded_addresses}
```

Comment:
`dict` instead of `Dict` since `from __future__ import annotations` is in effect?